lib/Dialect/AIEX/Transforms/AIECreateCores.cpp

```diff
@@ -173,7 +173,7 @@ struct AIECreateCoresPass : public AIECreateCoresBase<AIECreateCoresPass> {
           // the function argument's type is a scalar, we promote it to a
           // one-element memref, and do a load to the buffer at index 0
           for (auto pair : coreBufTypes) {
-            MemRefType t = pair.first;
+            [[maybe_unused]] MemRefType t = pair.first;
```

Comment:
This should be inside the if statement. Also, maybe consider using `pair.first.getShape()` in the asserts instead of capturing `pair.first` in `t`.

Reply:
Same as above.